public/request/ticket/create.php

```diff
@@ -1,17 +1,27 @@
 <?php
 
-require_once __DIR__ . '/../../../vendor/autoload.php';
-require_once __DIR__ . '/../../../lib/bootstrap.php';
+use App\Legacy\Models\User;
+use Illuminate\Support\Facades\Validator;
 
-if (!ValidatePOSTChars("ipm")) {
-    echo "FAILED";
-    exit;
+if (!authenticateFromCookie($user, $permissions, $userDetail)) {
+    return back()->withErrors(__('legacy.error.permissions'));
 }
 
-$achievementID = requestInputPost('i');
-$problemType = requestInputPost('p');
-$modeS = requestInputPost('m');
-$hardcore = ($modeS === "2") ? 1 : 0;
+/** @var User $user */
+$user = request()->user();
+
+$input = Validator::validate(request()->post(), [
+    'achievement' => 'required|integer|exists:mysql_legacy.Achievements,ID',
+    'mode' => 'required|boolean',
+    'issue' => 'required|integer|min:1|max:2',
+    'note' => 'sometimes|string|max:2',
+]);
+
+$achievementID = (int) $input['achievement'];
+$problemType = $input['issue'];
+$hardcore = (int) $input['mode'];
+
+// TODO untangle $_POST['note']
```

Comment:
is this related to the validation? it reads as if the note cannot be more than two characters long.

Reply:
Verified via test.

>  The note must not be greater than 2 characters.

Reply:
I was in the middle of rewriting the valdiation rules when I decided to push this branch anyways :) 
The validation rule made no sense yet nor did `note` being a nested form field, hence the TODO note.
Untangled here: https://github.com/RetroAchievements/RAWeb/commit/9acb69ad08d355edfc2f9fa2811dc5b8547b0602